econml/tests/test_treatment_featurization.py

```diff
@@ -0,0 +1,433 @@
+        poly_config_scikit['treatment_featurizer'] = PolynomialFeatures(degree=2, include_bias=False)
+        poly_config_scikit['squeeze_Ts'] = [False]
+
+        poly_IV_config = deepcopy(poly_config)
+        poly_IV_config['DGP_params']['d_z'] = 1
+        poly_IV_config['DGP_params']['nuisance_TZ'] = lambda Z: Z
+        poly_IV_config['est_dicts'] = [
+            {'class': OrthoIV, 'init_args': {'model_t_xwz': RandomForestRegressor(), 'projection': True}},
+            {'class': DMLIV, 'init_args': {'model_t_xwz': RandomForestRegressor()}},
+        ]
+
+        poly_1d_config = deepcopy(poly_config)
+        poly_1d_config['treatment_featurizer'] = polynomial_1d_treatment_featurizer
+        poly_1d_config['actual_cme'] = poly_1d_actual_cme
+        poly_1d_config['est_dicts'].append({
+            'class': NonParamDML,
+            'init_args': {
+                'model_y': LinearRegression(),
+                'model_t': LinearRegression(),
+                'model_final': StatsModelsLinearRegression()}})
+
+        poly_1d_IV_config = deepcopy(poly_IV_config)
+        poly_1d_IV_config['treatment_featurizer'] = polynomial_1d_treatment_featurizer
+        poly_1d_IV_config['actual_cme'] = poly_1d_actual_cme
+        poly_1d_IV_config['est_dicts'] = [
+            {'class': NonParamDMLIV, 'init_args': {'model_final': StatsModelsLinearRegression()}},
+            {'class': DRIV, 'init_args': {'fit_cate_intercept': True}},
+            {'class': LinearDRIV, 'init_args': {}},
+            {'class': SparseLinearDRIV, 'init_args': {}},
+            {'class': ForestDRIV, 'init_args': {}},
+        ]
+
+        configs = [
+            identity_config,
+            poly_config,
+            poly_config_scikit,
+            poly_IV_config,
+            poly_1d_config,
+            poly_1d_IV_config
+        ]
+
+        for config in configs:
+            for squeeze_Y in config['squeeze_Ys']:
+                for squeeze_T in config['squeeze_Ts']:
+                    config['DGP_params']['squeeze_Y'] = squeeze_Y
+                    config['DGP_params']['squeeze_T'] = squeeze_T
+                    dgp = DGP(**config['DGP_params'])
+                    data_dict = dgp.gen_data()
+                    Y = data_dict['Y']
+                    T = data_dict['T']
+                    X = data_dict['X']
+                    feat_T = config['treatment_featurizer'].fit_transform(T)
+
+                    est_dicts = config['est_dicts']
+
+                    for est_dict in est_dicts:
+                        estClass = est_dict['class']
+                        init_args = est_dict['init_args']
+                        init_args['treatment_featurizer'] = config['treatment_featurizer']
+
+                        est = estClass(**init_args)
+                        est.fit(**data_dict)
+
+                        #  test that treatment names are assigned for the featurized treatment
+                        assert (est.cate_treatment_names() is not None)
```

Comment:
Doesn't need to be in this test method, but tests should cover the `cate_treatment_names` method more thoroughly, verifying actual output names when the featurizer is and is not set, and when input names are and are not passed.

Reply:
Tests added